1 Tỉ lệ người sống sót là bao nhiêu : khoảng 50%
2 Trên tàu có bao nhiêu nữ : 40%
3 Có bao nhiêu khách là hạng sang : 10%
4 Lượng người có Người thân và một mình ai sẽ nhiều hơn : Người thân
5 Nơi lên tàu ở đâu đông nhất :Southampton
6 Đàn ông trưởng thành có nhiều vé vip hơn không : có
7 Vé nào đông người nhất : Third
8 Hạng nhất có sống nhiều hơn hạng ba không : không
9 Đối tượng nào lên tàu nhiều nhất : đàn ông
10 Người có gia đình và người độc thân ai chết nhiều hơn : độc thân

11 Mức giá vé (fare) cao nhất thuộc về khoang hạng (class) nào : Hạng nhất (First)
12 Độ tuổi (age) của nhóm hành khách trên tàu phần lớn là người già hay người trẻ : người trẻ (trung bình khoảng 29 - 30 tuổi)
13 Hành khách đi cùng anh chị em/vợ chồng (sibsp) hay đi cùng cha mẹ/con cái (parch) đông hơn : anh chị em/ vợ chồng (sibsp)
14 Ký hiệu boong tàu (deck) nào xuất hiện nhiều nhất trong danh sách dữ liệu : boong C
15 Hành khách lên tàu từ thành phố cảng (embark_town) nào phải trả mức giá vé (fare) trung bình cao nhất : Cherbourg

In [ ]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [8]:
titanic = sns.load_dataset('titanic')
df = pd.read_csv('Titanic-Dataset.csv')

5 dòng đầu và cuối của dataset

In [9]:
print("5 rows of the Titanic dataset:")
print(titanic.head())
print("\nLast 5 rows of the Titanic dataset:")
print(titanic.tail())

5 rows of the Titanic dataset:
   survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True  

Last 5 rows of the Titanic dataset:
     survived  pclass     sex   age  sibsp  parch   fare embarked   class  \
886         0       2    male  27.0      0      0  13.00        S  Second 

mô tả của dataset số liệu trung bình

In [6]:
print("\nSummary statistics of the Titanic dataset:")
print(titanic.describe())


Summary statistics of the Titanic dataset:
         survived      pclass         age       sibsp       parch        fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200


số cột và dòng của dataset

In [7]:
print("Shape of the Titanic dataset:")
print(titanic.shape)

Shape of the Titanic dataset:
(891, 15)


In [ ]:
print("Dataset information:")
print(titanic.info())
print("Missing values:")
print(titanic.isnull().sum())
print(titanic.isnull().sum())
most_null = titanic.isnull().sum().idxmax()

In [ ]:
print("Average age of passengers:")
print(titanic.describe()['age']['mean'])
print(f"Most ticket value: {df['Fare'].max()}")

In [ ]:
# 4. How many male and female passengers are there?
gender_counts = df['Sex'].value_counts()
print("Count of male and female passengers:")
print(gender_counts)

In [ ]:
# 5. Which passenger class (1, 2, 3) has the most passengers? Count and percentage.
class_counts = df['Pclass'].value_counts()
class_percentages = df['Pclass'].value_counts(normalize=True) * 100

print("Passenger count by class:")
print(class_counts)
print("\nPassenger percentage by class (%):")
print(class_percentages)

In [ ]:
# 6. Filter all female passengers. What percentage of the total do they represent?
female_passengers = df[df['Sex'] == 'female']
female_percentage = (len(female_passengers) / len(df)) * 100

print(f"Number of female passengers: {len(female_passengers)}")
print(f"Percentage of female passengers: {female_percentage:.2f}%")

In [ ]:
# 7. Filter 1st class passengers. What is their average fare?
first_class = df[df['Pclass'] == 1]
avg_fare_first_class = first_class['Fare'].mean()

print(f"Average fare for first class passengers: {avg_fare_first_class:.2f}")

In [ ]:
# 8.1. How many passengers are under 18 years old?
under_18 = df[df['Age'] < 18]
print(f"Number of passengers under 18: {len(under_18)}")

In [ ]:
# 8.2. How many passengers are over 60 years old?
over_60 = df[df['Age'] > 60]
print(f"Number of passengers over 60: {len(over_60)}")

Tỉ lệ sống sót toàn tàu

In [ ]:
survival_rate = df['Survived'].mean() * 100
print(f"Tỉ lệ sống sót toàn tàu: {survival_rate:.2f}%")

Giải thích ý nghĩa: Tỉ lệ sống sót chỉ đạt khoảng hơn 38%, cho thấy thảm họa chìm tàu Titanic là một tai nạn nghiêm trọng với tỉ lệ tử vong cao. Số người thiệt mạng chiếm hơn 60% tổng số hành khách.

Dùng groupby: tỉ lệ sống sót theo giới tính và theo hạng vé

In [ ]:
survival_by_sex_class = df.groupby(['Sex', 'Pclass'])['Survived'].mean() * 100
print("Tỉ lệ sống sót theo giới tính và hạng vé (%):")
print(survival_by_sex_class)

Nhận xét so sánh:
- Theo giới tính: Tỉ lệ sống sót của phụ nữ cao vượt trội so với nam giới ở mọi hạng vé. Điều này phản ánh quy tắc "Phụ nữ và trẻ em lên trước" trong quá trình cứu nạn.
- Theo hạng vé: Có sự chênh lệch rõ ràng về tỉ lệ sống sót giữa các hạng vé. Hành khách hạng 1 (khoang hành khách cao cấp) có tỉ lệ sống sót cao nhất, tiếp đến là hạng 2 và thấp nhất là hạng 3. Điều này có thể do vị trí khoang của hạng 1 nằm gần boong tàu và xuồng cứu sinh hơn.
- Kết hợp cả hai yếu tố, phụ nữ hạng 1 có tỉ lệ sống sót gần như tuyệt đối, trong khi nam giới hạng 3 có tỉ lệ sống sót thấp nhất.

Trả lời Phần A - Câu 1: Trực quan số lượng hành khách nam và nữ

In [ ]:
gender_counts = df['Sex'].value_counts()
print("Số lượng hành khách theo giới tính:")
print(gender_counts)

Giải thích kết quả: Dựa vào bảng phân bố, số lượng khách nam gấp gần đôi số lượng khách nữ. Nam giới là lực lượng hành khách lớn nhất tham gia trên chuyến đi.

Trả lời Phần A - Câu 2: Số hành khách theo hạng vé (Pclass)

In [ ]:
class_counts = df['Pclass'].value_counts()
print("Số lượng hành khách ở mỗi hạng vé:")
print(class_counts)

Giải thích kết quả: Nhóm khách sử dụng hạng vé số 3 (hạng vé bình dân) chiếm số lượng áp đảo so với các phân khúc hành khách khác. Số khách của hạng 1 và hạng 2 cộng lại vẫn chưa bắt kịp số lượng hành khách hạng 3. Điều này cho thấy tầng lớp lao động chiếm tỷ trọng lớn trong chuyến đi này.

Trả lời Phần A - Câu 3: Thống kê số lượng phần trăm hành khách nữ

In [ ]:
female_passengers = df[df['Sex'] == 'female']
female_percentage = (len(female_passengers) / len(df)) * 100
print(f"Tổng số phần trăm hành khách nữ: {female_percentage:.2f}%")

Giải thích kết quả: Nữ giới chỉ chiếm hơn 35.24% số người tham gia chuyến hành trình trên số liệu báo cáo, điều đó có thể lý giải vì sao nhóm hành khách có tử suất cao chủ yếu là nằm ở nam giới. Mật độ nam giới đông hơn.

Đối chiếu dự đoán 1: Tỉ lệ người sống sót là khoảng 50%

In [ ]:
actual_survival_rate = df['Survived'].mean() * 100
print(f"Tỉ lệ sống sót thực tế: {actual_survival_rate:.2f}%")

Kết luận: Sai. Tỉ lệ sống sót thực tế chỉ đạt khoảng 38.38%, thấp hơn đáng kể so với dự đoán. Thực tế thảm họa khốc liệt hơn dự báo 50% ban đầu.

Đối chiếu dự đoán 2: Trên tàu có khoảng 40% nữ giới

In [ ]:
actual_female_percentage = (df['Sex'] == 'female').mean() * 100
print(f"Tỉ lệ nữ giới thực tế: {actual_female_percentage:.2f}%")

Kết luận: Chưa hoàn toàn chính xác. Tỉ lệ nữ giới thực tế trên tàu là khoảng 35.24%. Con số này gần sát với mức dự đoán 40%, nhưng thực tế lượng nam giới vẫn đông hơn một chút so với dự tính.

Đối chiếu dự đoán 3: Có khoảng 10% khách là hạng sang (hạng 1)

In [ ]:
actual_class1_percentage = (df['Pclass'] == 1).mean() * 100
print(f"Tỉ lệ khách hạng sang (hạng 1) thực tế: {actual_class1_percentage:.2f}%")

Kết luận: Sai. Tỉ lệ hành khách hạng sang (hạng 1) thực tế lên tới hơn 24.24%, cao hơn gấp đôi so với mức dự kiến 10%. Điều này thể hiện rằng có một lượng tương đối lớn giới thượng lưu và người giàu tham gia trên chuyến đi này.